### PR TITLE
Loki integration

### PR DESCRIPTION
**Pull request by Abuelodelanada**
## Issue

This PR add Loki integration to Pushgateway

## Testing Instructions

1. Run itest

2. Deploy this bundle:
```yaml
bundle: kubernetes
applications:
  loki:
    charm: loki-k8s
    channel: edge
    revision: 99
    series: focal
    resources:
      loki-image: 82
    scale: 1
    constraints: arch=amd64
    storage:
      active-index-directory: kubernetes,1,1024M
      loki-chunks: kubernetes,1,1024M
    trust: true
  pushgateway:
    charm: ../../prometheus-pushgateway-k8s_ubuntu-22.04-amd64.charm
    scale: 1
    constraints: arch=amd64
    resources:
      pushgateway-image: ghcr.io/canonical/prometheus-pushgateway:dev
    storage:
      pushgateway-store: kubernetes,1,1024M
    trust: true
relations:
- - pushgateway:log-proxy
  - loki:logging
```

... and check there are logs in Loki:

```shell
curl -G "http://LOKI_IP:3100/loki/api/v1/query_range" --data-urlencode 'query={juju_charm="prometheus-pushgateway-k8s"} |= ``' 
```

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -93,6 +95,10 @@ def __init__(self, *args):
                 self.on.pushgateway_pebble_ready,
             ],
         )
+        # Enable log forwarding for Loki and other charms that implement loki_push_api
+        self._logging = LogProxyConsumer(
+            self, relation_name="log-proxy", log_files=[PUSHGATEWAY_LOG]
```
Comment (PietroPasotti) [inline src/charm.py:100]: don't we call this `logging` on all other charms?

File: src/charm.py [hunk 2]
```diff
@@ -117,7 +123,7 @@ def _command(self) -> str:
         if self._web_config:
             args.append(f"--web.config.file={WEB_CONFIG_PATH}")
 
-        command = [PUSHGATEWAY_BINARY] + args
+        command = [f"/bin/sh -c '{PUSHGATEWAY_BINARY}"] + args + [f"2>&1 | tee {PUSHGATEWAY_LOG}'"]
```
Comment (PietroPasotti) [inline src/charm.py:126]: perhaps add a comment to explain this (obscure) shell sorcery?

**Review by PietroPasotti — CHANGES_REQUESTED** [review]

File: metadata.yaml
[the remark this answers is not in the document]
Comment (Abuelodelanada) [reply metadata.yaml:1]: Nop,

Ingress was added in [this PR](https://github.com/canonical/prometheus-pushgateway-k8s-operator/pull/16) and [TLS in this one](https://github.com/canonical/prometheus-pushgateway-k8s-operator/pull/15)

File: src/charm.py (hunk 2 above)
Comment (Abuelodelanada) [reply src/charm.py:135]: Done

File: src/charm.py (hunk 1 above)
Comment (Abuelodelanada) [reply src/charm.py:100]: Yes, that's right. We use `log-proxy`

**Review by sed-i — APPROVED** [review]

**Pull request merged**